apps/storefront/src/utils/headlessInitializer.ts

```diff
@@ -0,0 +1,79 @@
+import { getAPIBaseURL } from '../shared/service/request/base';
+import { Environment } from '../types';
+
+const BUYER_PORTAL_INJECTED_SCRIPT_CLASS = `buyer-portal-scripts-headless`;
+
+export async function initHeadlessScripts() {
+  const parseAndInsertStorefrontScripts = (storefrontScripts: string) => {
+    const b2bScriptDocument = new DOMParser().parseFromString(storefrontScripts, 'text/html');
+    const b2bScriptNodes = b2bScriptDocument.querySelectorAll('script');
+
+    if (b2bScriptNodes.length > 0) {
+      const existingBuyerPortalScriptNodes = document.querySelectorAll(
+        `script.${BUYER_PORTAL_INJECTED_SCRIPT_CLASS}`,
+      );
+      if (existingBuyerPortalScriptNodes.length > 0) {
+        existingBuyerPortalScriptNodes.forEach((oldNode) => {
+          oldNode.parentNode?.removeChild(oldNode);
+        });
+      }
+
+      b2bScriptNodes.forEach((scriptNode) => {
+        const newScriptElement = document.importNode(scriptNode, true);
+        newScriptElement.className = BUYER_PORTAL_INJECTED_SCRIPT_CLASS;
+
+        document.body.appendChild(newScriptElement);
+      });
+    }
+  };
+
+  async function getScriptContent(originUrl: string): Promise<string> {
+    // cSpell:ignore storehash, channelid
+    const headlessScriptNode: HTMLElement | null = document.querySelector(
+      'script[data-storehash][data-channelid]',
+    );
+    const params: {
+      siteUrl: string;
+      storeHash: string;
+      channelId: string;
+      environment?: Environment;
+    } = {
+      siteUrl: originUrl,
+      storeHash: headlessScriptNode?.dataset?.storehash ?? '',
+      channelId: headlessScriptNode?.dataset?.channelid ?? '',
+      environment: headlessScriptNode?.dataset?.environment as Environment,
```

Comment:
Added ability to set `environment` via the script tag when including the headless script.